contracts/truefi2/DebtToken.sol

```diff
@@ -4,37 +4,108 @@ pragma solidity 0.6.10;
 import {ERC20} from "../common/UpgradeableERC20.sol";
 import {IDebtToken} from "./interface/ILoanToken2.sol";
 import {ITrueFiPool2} from "./interface/ITrueFiPool2.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
 
 contract DebtToken is IDebtToken, ERC20 {
-    function borrower() external override view returns (address) {}
-
-    function amount() external override view returns (uint256) {}
-
-    function debt() external override view returns (uint256) {}
-
-    function pool() external override view returns (ITrueFiPool2) {}
-
-    function status() external override view returns (Status) {}
-
-    function redeem(uint256 _amount) external override {}
-
-    function repay(address _sender, uint256 _amount) external override {}
-
-    function repayInFull(address _sender) external override {}
-
-    function reclaim() external override {}
-
-    function liquidate() external override {}
-
-    function repaid() external override view returns (uint256) {}
-
-    function isRepaid() external override view returns (bool) {}
-
-    function balance() external override view returns (uint256) {}
-
-    function value(uint256 _balance) external override view returns (uint256) {}
-
-    function token() external override view returns (ERC20) {}
-
-    function version() external override pure returns (uint8) {}
+    using SafeMath for uint256;
+    using SafeERC20 for ERC20;
+
+    address public override borrower;
+    uint256 public override debt;
+
+    address public liquidator;
+    ITrueFiPool2 public override pool;
+    ERC20 public override token;
+
+    uint256 public redeemed;
+
+    Status public override status;
+
+    /**
+     * @dev Emitted when a DebtToken is redeemed for underlying tokens
+     * @param receiver Receiver of tokens
+     * @param burnedAmount Amount of DebtTokens burned
+     * @param redeemedAmount Amount of token received
+     */
+    event Redeemed(address receiver, uint256 burnedAmount, uint256 redeemedAmount);
+
+    /**
+     * @dev Emitted when loan gets liquidated
+     * @param status Final loan status
+     */
+    event Liquidated(Status status);
+
+    function initialize(
+        ITrueFiPool2 _pool,
+        address _lender,
+        address _borrower,
+        address _liquidator,
+        uint256 _debt
+    ) external initializer {
+        ERC20.__ERC20_initialize("TrueFi Debt Token", "DEBT");
+
+        pool = _pool;
+        token = _pool.token();
+        borrower = _borrower;
+        liquidator = _liquidator;
+        debt = _debt;
+        status = Status.Defaulted;
+        _mint(_lender, _debt);
+    }
+
+    /**
+     * @dev Redeem DebtToken balances for underlying token
+     * Can only call this function after the loan is Closed
+     * @param _amount amount to redeem
+     */
+    function redeem(uint256 _amount) external override {
```

Comment:
We should still check `Status.Defaulted` || `Status.Liquidated` (possibly even `Status.Settled` if we're planning to have LoanToken2 inherit from DebtToken). I'm okay with inlining the check though.

Reply:
DebtToken can have only 2 states now. I'd wait with adding any LoanToken-specific functionality to when we start deprecating LTs and follow YAGNI principle.

Reply:
In general I agree with YAGNI, but in this case IDebtToken still has more than two possible states -- we plan to deprecate the others, but they haven't even been marked for deprecation yet.

It might happen to be that only two states are reachable in current code, but it would be easy to accidentally reintroduce the other states without noticing `redeem()` would need to guard against them. As long as it's possible for this enum to have unexpected values (i.e., 0-3), I think we should protect from those values.

For now, perhaps we could add the require and prepend with a comment, something like this?
```
// TODO remove the require checks when Awaiting/Funded/Withdrawn enum values are removed from IDebtToken
require(status() >= 4, "error log")
```

As soon as we remove the other statuses from the enum, we should notice test failures, see the TODO here, and be ready to remove these two lines.